### PR TITLE
[Test] Add unit tests for srt/managers/scheduler_recv_skipper.py and  srt/managers/scheduler_input_blocker.py

### PR DESCRIPTION
**Pull request by evanderfff123-boop**
28 test cases covering SchedulerRecvSkipper and SchedulerInputBlocker.

No server, no GPU, no model loading.
Ref: #20865

<!-- Thank you for your contribution! Please follow these guidelines to enhance your pull request. If anything is unclear, submit your PR and reach out to maintainers for assistance. Join our Slack community at https://slack.sglang.io to discuss further. -->

## Motivation
Added two new test files: `test/registered/unit/managers/test_scheduler_recv_skipper.py` and `test_scheduler_input_blocker.py`

`managers/scheduler_recv_skipper.py` and `managers/scheduler_input_blocker.py` currently have no dedicated unit tests. The recv skipper contains a weighted-counter state machine that throttles how often the scheduler checks for new requests under different forward modes (DECODE, TARGET_VERIFY, etc.). The input blocker implements a three-state gate (UNBLOCKED → BLOCKED → GLOBAL_UNBLOCK_BARRIER → UNBLOCKED) that atomically blocks request flow while tokenizer performs large-batch packaging, coordinated across distributed ranks via a poll-based barrier. Both can regress silently. This PR adds CPU-only coverage for the current behavior.

**Coverage**

**`scheduler_recv_skipper.py`** (10 tests):
- `maybe_create` — interval ≤ 1 returns None, interval > 1 creates instance, DP attention asserts.
- `handle` — counter accumulates across calls, crosses threshold after N calls, resets after trigger, per-mode weights (DECODE=2 triggers faster, TARGET_VERIFY=4 triggers immediately, None=4 triggers immediately), zero weight never triggers (100 iterations), unknown forward mode falls back to default weight.

**`scheduler_input_blocker.py`** (18 tests):
- Core state machine — initial UNBLOCKED, plain requests pass through, BLOCK transitions, requests queue when blocked, double-block asserts, unblock-without-block asserts, unblock transitions to GLOBAL_UNBLOCK_BARRIER.
- Barrier coordination — `local_arrive()` called on unblock, barrier-not-arrived leaves pending intact, barrier-arrived drains pending + resets to UNBLOCKED, full cycle restores normal pass-through.
- Noop mode — accepts `None`, rejects non-None, still polls barrier.
- Guard region context manager — sends BLOCK before yield + UNBLOCK after, sends UNBLOCK even on exception.
- Pending requests preserve insertion order.

The tests are CPU-only and do not start a server or load a model. Mock usage is limited to `PollBasedBarrier` (requires `torch.distributed`) and `os.environ` (for env-var-driven weights). All state machine assertions operate on real `SchedulerRecvSkipper` / `SchedulerInputBlocker` instances. They are registered via:
<!-- Describe the purpose and goals of this pull request. -->

```python
# scheduler_recv_skipper
register_cpu_ci(est_time=4, suite="base-a-test-cpu")
register_cpu_ci(est_time=5, suite="base-b-test-cpu")

# scheduler_input_blocker
register_cpu_ci(est_time=5, suite="base-a-test-cpu")
register_cpu_ci(est_time=6, suite="base-b-test-cpu")
```

**Test plan:**
test_scheduler_recv_skipper.py:
```shell
pytest test/registered/unit/managers/test_scheduler_recv_skipper.py -v
==================================================================================================== test session starts =====================================================================================================
platform linux -- Python 3.12.13, pytest-9.0.3, pluggy-1.6.0 -- /home/evanderf/sglang/.venv/bin/python3.12
cachedir: .pytest_cache
rootdir: /home/evanderf/sglang/test
configfile: pytest.ini
plugins: anyio-4.13.0, asyncio-1.4.0, cov-7.0.0
asyncio: mode=Mode.AUTO, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 10 items                                                                                                                                                                                                           

test/registered/unit/managers/test_scheduler_recv_skipper.py::TestMaybeCreate::test_enable_dp_attention_asserts PASSED                                                                                                 [ 10%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestMaybeCreate::test_interval_gt_1_creates_instance PASSED                                                                                              [ 20%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestMaybeCreate::test_interval_le_1_returns_none PASSED                                                                                                  [ 30%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestHandle::test_counter_resets_after_trigger PASSED                                                                                                     [ 40%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestHandle::test_decode_weight_2_triggers_faster PASSED                                                                                                  [ 50%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestHandle::test_four_calls_trigger_on_threshold_4 PASSED                                                                                                [ 60%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestHandle::test_none_forward_mode_uses_own_weight PASSED                                                                                                [ 70%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestHandle::test_target_verify_uses_own_weight PASSED                                                                                                    [ 80%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestHandle::test_unknown_forward_mode_falls_back_to_default PASSED                                                                                       [ 90%]
test/registered/unit/managers/test_scheduler_recv_skipper.py::TestHandle::test_zero_weight_never_triggers PASSED                                                                                                       [100%]

====================================================================================================== warnings summary ======================================================================================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================================== 10 passed, 2 warnings in 4.41s ===============================================================================================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```
test_scheduler_input_blocker.py:
```shell
pytest test/registered/unit/managers/test_scheduler_input_blocker.py -v
==================================================================================================== test session starts =====================================================================================================
platform linux -- Python 3.12.13, pytest-9.0.3, pluggy-1.6.0 -- /home/evanderf/sglang/.venv/bin/python3.12
cachedir: .pytest_cache
rootdir: /home/evanderf/sglang/test
configfile: pytest.ini
plugins: anyio-4.13.0, asyncio-1.4.0, cov-7.0.0
asyncio: mode=Mode.AUTO, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 18 items                                                                                                                                                                                                           

test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_barrier_arrived_drains_pending_requests PASSED                                                                          [  5%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_barrier_arrived_resets_state_to_unblocked PASSED                                                                        [ 11%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_barrier_not_arrived_does_not_drain PASSED                                                                               [ 16%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_block_req_changes_state_to_blocked PASSED                                                                               [ 22%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_double_block_asserts PASSED                                                                                             [ 27%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_full_cycle_restores_normal_operation PASSED                                                                             [ 33%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_initial_state_is_unblocked PASSED                                                                                       [ 38%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_noop_accepts_none PASSED                                                                                                [ 44%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_noop_rejects_non_none PASSED                                                                                            [ 50%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_noop_still_polls_barrier PASSED                                                                                         [ 55%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_pending_reqs_preserve_insertion_order PASSED                                                                            [ 61%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_plain_request_passes_through_when_unblocked PASSED                                                                      [ 66%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_requests_are_queued_when_blocked PASSED                                                                                 [ 72%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_unblock_calls_local_arrive PASSED                                                                                       [ 77%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_unblock_transitions_to_barrier_state PASSED                                                                             [ 83%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestSchedulerInputBlocker::test_unblock_without_block_asserts PASSED                                                                                    [ 88%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestInputBlockerGuardRegion::test_sends_block_before_yield_and_unblock_after PASSED                                                                     [ 94%]
test/registered/unit/managers/test_scheduler_input_blocker.py::TestInputBlockerGuardRegion::test_sends_unblock_even_on_exception PASSED                                                                                [100%]

====================================================================================================== warnings summary ======================================================================================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: 14 warnings
  /home/evanderf/sglang/.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================== 18 passed, 16 warnings in 5.61s ===============================================================================================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```

## Accuracy Tests
N/A — this is a test-only change. No kernel or model-forward code is touched.
<!-- If this pull request affects model outputs (e.g., changes to the kernel or model forward code), provide accuracy test results. -->

## Speed Tests and Profiling
N/A — this PR only adds unit tests and does not affect the inference path.
<!-- If this pull request impacts inference speed, provide benchmarking and profiling results. -->

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.





























<!-- pr-states:start -->
---
### CI States

Latest PR Test (Base): <!-- slot:pr-test:start -->:x: [Run #27545923624](https://github.com/sgl-project/sglang/actions/runs/27545923624)<!-- slot:pr-test:end -->
Latest PR Test (Extra): <!-- slot:pr-test-extra:start -->:x: [Run #27545923666](https://github.com/sgl-project/sglang/actions/runs/27545923666)<!-- slot:pr-test-extra:end -->
<!-- pr-states:end -->

### CODE REVIEW

**Comment by evanderfff123-boop** [comment]
> ## Code Review
> This pull request adds unit tests for `SchedulerInputBlocker` and `SchedulerRecvSkipper` to verify their state transitions, request queueing, and behavior under different forward modes and weights. The review feedback suggests strengthening the assertions in the guard region tests by verifying the exact sequence of mock calls rather than just checking call counts or the final call's arguments.
> 
> Important
> 
> The [consumer version of Gemini Code Assist on GitHub](https://developers.google.com/gemini-code-assist/docs/review-repo-code) is being sunset. Starting **June 18, 2026**, new organization installations will be blocked, and all code review activity will officially cease on **July 17, 2026**. For more details on the timeline and next steps, please review the [Help Documentation](https://developers.google.com/gemini-code-assist/docs/deprecations/consumer-code-review).

Thanks. The guard region tests have been updated to assert the exact `send_pyobj` call sequence (`BLOCK` followed by `UNBLOCK`) for both the normal path and the exception path, instead of only checking call counts or the final call.
